riff-raff/app/notification/DeployFailureNotifications.scala

```diff
@@ -50,46 +55,90 @@ class DeployFailureNotifications(config: Config,
     }
   }
 
-  def failedDeployNotification(uuid: Option[UUID], parameters: DeployParameters): Unit = {
-
-    val deriveAnghammaradTargets = for {
-      yaml <- targetResolver.fetchYaml(parameters.build)
-      deployGraph <- Resolver.resolveDeploymentGraph(yaml, deploymentTypes, magenta.input.All).toEither
-    } yield {
-      TargetResolver.extractTargets(deployGraph).toList.flatMap { target =>
-        val maybeAccountId = uuid.flatMap(uuid => getAwsAccountIdTarget(target, parameters, uuid)).toList
-        List(App(target.app), Stack(target.stack), Stage(parameters.stage.name)) ++ maybeAccountId
+  def getTargets(uuid: Option[UUID], maybeParameters: Option[DeployParameters]): List[Target] = {
+    val attemptToDeriveTargets = maybeParameters.map { parameters =>
+       for {
+        yaml <- targetResolver.fetchYaml(parameters.build)
+        deployGraph <- Resolver.resolveDeploymentGraph(yaml, deploymentTypes, magenta.input.All).toEither
+      } yield {
+        TargetResolver.extractTargets(deployGraph).toList.flatMap { target =>
+          val maybeAccountId = uuid.flatMap(uuid => getAwsAccountIdTarget(target, parameters, uuid)).toList
+          List(App(target.app), Stack(target.stack), Stage(parameters.stage.name)) ++ maybeAccountId
+        }
       }
     }
+    attemptToDeriveTargets match {
+      case Some(Right(targets)) => targets
+      case _ => riffRaffTargets
```

Comment:
This is new behaviour, previously if we couldn't find the targets we just dropped the notification entirely: https://github.com/guardian/riff-raff/pull/614/files#diff-0dceec64b87cbc45ffb3aabb2b6f1fb0abd9b04fcc1e1cde546cd01b49275362L80-L81